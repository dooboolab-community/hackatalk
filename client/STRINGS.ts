import * as Localization from 'expo-localization';

import en from './assets/langs/en.json';
import i18n from 'i18n-js';
import ko from './assets/langs/ko.json';

i18n.fallbacks = true;
i18n.translations = { en, ko };
i18n.locale = Localization.locale;

export const getString = (param: string, mapObj?: Record<string, unknown>): string => {
  if (mapObj) {
    return i18n.t(param, mapObj);
  }

  return i18n.t(param);
};
