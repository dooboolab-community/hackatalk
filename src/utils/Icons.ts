import icAddW from '../../assets/icons/btnAdd.png';
import icBack from '../../assets/icons/btnBack.png';
import icCamera from '../../assets/icons/camera.png';
import icCircleX from '../../assets/icons/x_circle.png';
import icError from '../../assets/icons/error.png';
import icIcon from '../../assets/icons/icon.png';
import icLogoD from '../../assets/icons/logo_d.png';
import icLogoW from '../../assets/icons/logo_w.png';
import icPic from '../../assets/icons/pictoGallery.png';
import icProfile from '../../assets/icons/profile.png';
import icProfileW from '../../assets/icons/profile_w.png';
import icSearch from '../../assets/icons/search.png';
import icSearchW from '../../assets/icons/search_w.png';
import icSettingW from '../../assets/icons/setting_w.png';
import icSmile from '../../assets/icons/icSmile.png';
import svgApple from '../../assets/icons/apple.svg';
import svgFacebook from '../../assets/icons/facebook.svg';
import svgGoogle from '../../assets/icons/google.svg';
import svgPlus from '../../assets/icons/plus.svg';

export const IC_ADD_W = icAddW;
export const IC_ICON = icIcon;
export const IC_CAMERA = icCamera;
export const IC_CIRCLE_X = icCircleX;
export const IC_ERROR = icError;
export const IC_PROFILE = icProfile;
export const IC_BACK = icBack;
export const IC_PIC = icPic;
export const IC_SEARCH = icSearch;
export const IC_SMILE = icSmile;
export const IC_LOGO_D = icLogoD;
export const IC_LOGO_W = icLogoW;
export const IC_SETTING_W = icSettingW;
export const IC_SEARCH_W = icSearchW;
export const IC_PROFILE_W = icProfileW;
export const SvgApple = svgApple;
export const SvgGoogle = svgGoogle;
export const SvgFacebook = svgFacebook;
export const SvgPlus = svgPlus;

export default [
  // Do not run svgs for now
  // SvgApple,
  // SvgGoogle,
  // SvgFacebook,
  // SvgPlus,
  // SvgSetting,
  // SvgPerson,
  // SvgSearch,
  IC_ADD_W,
  IC_ICON,
  IC_CAMERA,
  IC_CIRCLE_X,
  IC_ERROR,
  IC_PROFILE,
  IC_BACK,
  IC_PIC,
  IC_SEARCH,
  IC_SMILE,
  IC_LOGO_D,
  IC_LOGO_W,
  IC_SETTING_W,
  IC_SEARCH_W,
  IC_PROFILE_W,
];
