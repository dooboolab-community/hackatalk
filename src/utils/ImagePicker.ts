import * as ImagePicker from 'expo-image-picker';
import * as Permissions from 'expo-permissions';

const photoOptions = {
  mediaTypes: ImagePicker.MediaTypeOptions.Images,
  allowsEditing: true,
  exif: true,
};

const requestPermissions = async (
  type: string,
): Promise<Permissions.PermissionStatus> => {
  if (type === 'photo') {
    const { status } = await Permissions.askAsync(Permissions.CAMERA_ROLL);
    return status;
  }
  const { status } = await Permissions.askAsync(Permissions.CAMERA);
  return status;
};

export const launchCameraAsync = async (): Promise<ImagePicker.ImagePickerResult | null> => {
  const permissionStatus = await requestPermissions('camera');
  if (permissionStatus === Permissions.PermissionStatus.GRANTED) {
    const result = await ImagePicker.launchCameraAsync(photoOptions);
    return result;
  }
  return null;
};

export const launchImageLibraryAsync = async (): Promise<ImagePicker.ImagePickerResult | null> => {
  const permissionStatus = await requestPermissions('photo');
  if (permissionStatus === Permissions.PermissionStatus.GRANTED) {
    const result = await ImagePicker.launchImageLibraryAsync(photoOptions);
    return result;
  }
  return null;
};
