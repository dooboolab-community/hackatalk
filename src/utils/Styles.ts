import { Dimensions } from 'react-native';

const { width, height } = Dimensions.get('window');
let calRatio = width <= height ? 16 * (width / height) : 16 * (height / width);

if (width <= height) {
  if (calRatio < 9) {
    calRatio = width / 9;
  } else {
    calRatio = height / 18;
  }
} else {
  if (calRatio < 9) {
    calRatio = height / 9;
  } else {
    calRatio = width / 18;
  }
}

export const screenWidth = width;
export const screenHeight = height;
export const ratio = calRatio / (360 / 9);
