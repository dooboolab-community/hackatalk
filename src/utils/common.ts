export const validateEmail = (email: string): boolean => {
  const re = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
  return re.test(email);
};

export const validatePassword = (password: string): boolean => {
  const re = /^.*(?=.{6,15})(?=.*[0-9])(?=.*[a-zA-Z])(?=.*[!@#$%^&+=]).*$/;
  return re.test(password);
};
