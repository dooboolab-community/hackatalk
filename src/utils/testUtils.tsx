import React, { ReactElement } from 'react';

import RootProvider from '../providers';
import { ThemeType } from '../types';

export const createTestElement = (
  child: ReactElement,
  themeType?: ThemeType,
): ReactElement => (
  <RootProvider initialThemeType={themeType}>{child}</RootProvider>
);

export const createTestProps = (
  obj: object = {},
  moreScreenProps?: object,
): object | unknown | any => ({
  navigation: {
    navigate: jest.fn(),
    goBack: jest.fn(),
    resetRoot: jest.fn(),
  },
  screenProps: {
    changeThemeType: jest.fn(),
    ...moreScreenProps,
  },
  ...obj,
});
