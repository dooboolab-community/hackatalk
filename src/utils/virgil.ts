import { EThree, ICard, IPrivateKey } from '@virgilsecurity/e3kit-native';

import AsyncStorage from '@react-native-community/async-storage';
import Config from 'react-native-config';

const { ROOT_URL } = Config;

export let eThree: EThree;

/**
 * Virgil Pythia Service has a rate limit of 2 seconds.
 */
const SERVICE_DELAY = 2000;

/**
 * Return a promise that will be resolved after `ms` milliseconds.
 * @param {number} ms - Timeout in milliseconds.
 */
async function sleep(ms: number): Promise<void> {
  return new Promise((resolve) => {
    setTimeout(() => {
      resolve();
    }, ms);
  });
}

/**
 * Create a new user in the Virgil Cloud.
 * @param {EThree} e3kit - Instance of EThree class for current user.
 * @param {string} password - User's password.
 */
async function createUser(e3kit: EThree, password: string): Promise<void> {
  try {
    await e3kit.register();
    await e3kit.backupPrivateKey(password);
  } catch (error) {
    if (error.name === ' CloudEntryExistsError') {
      await e3kit.resetPrivateKeyBackup();
      sleep(SERVICE_DELAY);
      await e3kit.backupPrivateKey(password);
    } else {
      throw error;
    }
  }
}

/**
 * Returns true if private key's identifier matches the card's public key identifier.
 * @param {EThree} e3kit - Instance of EThree class for current user.
 * @param {ICard} card - User's Virgil Card.
 */
async function isKeyValid(e3kit: EThree, card: ICard): Promise<boolean> {
  const keyEntry = await e3kit.keyEntryStorage.load(e3kit.identity);

  if (!keyEntry) {
    return false;
  }
  const privateKey: IPrivateKey = e3kit.virgilCrypto.importPrivateKey(keyEntry.value);
  if (privateKey.identifier.byteLength !== card.publicKey.identifier.byteLength) {
    return false;
  }
  for (let i = 0; i < privateKey.identifier.byteLength; i += 1) {
    if (privateKey.identifier[i] !== card.publicKey.identifier[i]) {
      return false;
    }
  }
  return true;
}

/**
 * Rotate user's private key and save it in the Virgil Cloud.
 * @param {EThree} e3kit - Instance of EThree class for current user.
 * @param {string} password - User's password.
 */
async function rotateKey(e3kit: EThree, password: string): Promise<void> {
  await e3kit.cleanup();
  await e3kit.rotatePrivateKey();
  await e3kit.backupPrivateKey(password);
}

export const getTokenFactory = (identity: string): () => Promise<string> => {
  return (): Promise<string> =>
    fetch(`${ROOT_URL}/virgil-jwt/${encodeURIComponent(identity)}`)
      .then((res) => res.text());
};

/**
 * Restore user from the Virgil Cloud.
 * @param {EThree} e3kit - Instance of EThree class for current user.
 * @param {ICard} card - User's Virgil Card.
 * @param {string} password - User's password.
 */
async function restoreUser(e3kit: EThree, card: ICard, password: string): Promise<void> {
  try {
    await e3kit.restorePrivateKey(password);
    const keyValid = await isKeyValid(e3kit, card);
    if (!keyValid) {
      await e3kit.resetPrivateKeyBackup();
      await sleep(SERVICE_DELAY);
      await rotateKey(e3kit, password);
    }
  } catch (error) {
    if (error.name === 'PrivateKeyNoBackupError') {
      await sleep(SERVICE_DELAY);
      await rotateKey(e3kit, password);
    } else {
      throw error;
    }
  }
}

/**
 * Initialize user on current device.
 * @param {EThree} e3kit - Instance of EThree class for current user.
 * @param {string} password - User's password.
 */
export async function initUser(e3kit: EThree, password: string): Promise<void> {
  try {
    await e3kit.cleanup();
    const card = await e3kit.findUsers(e3kit.identity);
    await restoreUser(e3kit, card, password);
  } catch (error) {
    if (error.name === 'UsersNotFoundError') {
      await createUser(e3kit, password);
    } else {
      throw error;
    }
  }
}

export const encryptMessage = async (users: string[]): Promise<string | Buffer> => {
  if (!eThree) {
    throw new Error('eThree is not registered');
  }

  const receiversCard = await eThree.findUsers(users);
  const encryptedMessage = await eThree.encrypt('Hello Receivers!', receiversCard);

  return encryptedMessage;
};

export const decryptMessage = async (users: string[], encryptedMessage: string):
Promise<string | Buffer> => {
  if (!eThree) {
    throw new Error('eThree is not registered');
  }

  const sendersCard = await eThree.findUsers(users);

  const decryptedMessage = await eThree.decrypt(
    encryptedMessage,
    sendersCard[eThree.identity],
  );

  return decryptedMessage;
};

export const initializeEThree = async (userId: string): Promise<void> => {
  const getToken = getTokenFactory(userId);

  try {
    // @ts-ignore ==> https://github.com/VirgilSecurity/virgil-e3kit-js/issues/82
    eThree = await EThree.initialize(getToken, { AsyncStorage });
    const password = await AsyncStorage.getItem('password');
    await initUser(eThree, userId + password);
    AsyncStorage.removeItem('password');
  } catch (err) {
    console.log('Ethree register error', err);
  }
};
