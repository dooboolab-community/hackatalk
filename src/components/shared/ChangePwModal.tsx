import { Alert, KeyboardAvoidingView } from 'react-native';
import { DefaultTheme, ThemeProps, withTheme } from 'styled-components/native';
import { InnerContainer, ModalCloseButton, ModalHeader, ModalTitle, StyledTextInput } from '../screen/Setting/styles';
import React, { ReactElement, forwardRef, useImperativeHandle, useRef, useState } from 'react';

import Button from './Button';
import { Ionicons } from '@expo/vector-icons';
import Modal from 'react-native-modalbox';
import { SafeAreaView } from 'react-navigation';
import { getString } from '../../../STRINGS';
import { useThemeContext } from '../../providers/ThemeProvider';

export interface ChangPwModalRef {
  open: () => void;
  close: () => void;
}

interface Props {
  close?: () => void;
}

export function ChangePw(props: Props): ReactElement {
  const { theme } = useThemeContext();
  const [isValidCurrentPw, updateCurrentPwValid] = useState(false);
  const [currentPw, setCurrentPw] = useState('');
  const [newPw, setNewPw] = useState('');
  const [validationWord, setValidationWord] = useState('');
  const close = (): void => {
    updateCurrentPwValid(false);
    props.close && props.close();
  };
  const validateCurrent = (): void => {
    try {
      if (currentPw === 'right') {
        updateCurrentPwValid(true);
      } else {
        throw Error(getString('PASSWORD_MUST_MATCH'));
      }
    } catch (err) {
      Alert.alert('', err.message);
    };
  };
  const changePassword = async (): Promise<void> => {
    if (newPw === validationWord) {
      // TODO change password api call
      Alert.alert('', 'Password changed.', [
        {
          text: getString('OK'),
          onPress: (): void => {
            close();
          },
        },
      ]);
    } else {
      Alert.alert(getString('PASSWORD_MUST_MATCH'));
    }
  };
  return (
    <SafeAreaView style={{ flex: 1 }}>
      <KeyboardAvoidingView style={{ flex: 1 }} behavior="height" enabled>
        <ModalHeader>
          <ModalTitle>{getString('PASSWORD_CHANGE')}</ModalTitle>
          <ModalCloseButton
            testID="closeBtn"
            onPress={close}>
            <Ionicons name="md-close" size={32} color={theme.fontColor} />
          </ModalCloseButton>
        </ModalHeader>
        {
          isValidCurrentPw
            ? <InnerContainer>
              <StyledTextInput
                key="newPwTextInput"
                testID="newPwTextInput"
                isPassword
                onTextChanged={(pw): void => setNewPw(pw)}
                txtLabel={getString('PASSWORD_NEW')} />
              <StyledTextInput
                key="validationWordTextInput"
                testID="validationWordTextInput"
                isPassword
                onTextChanged={(pw): void => setValidationWord(pw)}
                txtLabel={getString('PASSWORD_NEW_REPEAT')} />
            </InnerContainer>
            : <InnerContainer>
              <StyledTextInput
                key="currentPwTextInput"
                testID="currentPwTextInput"
                isPassword
                onTextChanged={(pw): void => setCurrentPw(pw)}
                txtLabel={getString('PASSWORD_CURRENT')} />
            </InnerContainer>
        }
        <Button
          testID="checkCurrentPwBtn"
          onPress={isValidCurrentPw ? changePassword : validateCurrent}>
          {isValidCurrentPw ? getString('CONFIRM') : getString('NEXT')}
        </Button>
      </KeyboardAvoidingView>
    </SafeAreaView>
  );
}

const ModalWrapper = forwardRef<ChangPwModalRef, ThemeProps<DefaultTheme>>((props, ref) => {
  const modal: React.MutableRefObject<Modal | null> = useRef(null);
  const { theme } = useThemeContext();

  const open = (): void => {
    modal.current && modal.current.open();
  };

  const close = (): void => {
    modal.current && modal.current.close();
  };

  useImperativeHandle(ref, () => ({
    open,
    close,
  }));

  return (
    <Modal
      ref={modal}
      keyboardTopOffset={0}
      coverScreen
      backButtonClose
      style={{ backgroundColor: theme.background }}>

    </Modal>
  );
});

export default withTheme(ModalWrapper);
