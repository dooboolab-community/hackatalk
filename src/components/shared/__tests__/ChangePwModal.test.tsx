import * as React from 'react';
import ChangePwModal, { ChangePw } from '../ChangePwModal';

import {
  RenderResult,
  cleanup,
  fireEvent,
  render,
  wait,
} from '@testing-library/react-native';
import { createTestElement, createTestProps } from '../../../utils/testUtils';
import { getString } from '../../../../STRINGS';
import renderer from 'react-test-renderer';

let component: React.ReactElement;
let innerComponent: React.ReactElement;

describe('[ChangePw] screen', () => {
  let testingLib: RenderResult;

  beforeEach(() => {
    component = createTestElement(
      <ChangePwModal />
    );
    innerComponent = createTestElement(
      <ChangePw close={jest.fn()} />
    );
    testingLib = render(innerComponent);
  });

  it('renders without crashing', () => {
    const rendered = renderer.create(component).toJSON();
    expect(rendered).toMatchSnapshot();
    expect(rendered).toBeTruthy();

    const renderedContent = renderer.create(innerComponent).toJSON();
    expect(renderedContent).toMatchSnapshot();
    expect(renderedContent).toBeTruthy();
  });

  it('snapshot check if modal open', () => {
    expect(testingLib.asJSON()).toMatchSnapshot();
  });

  describe('interactions', () => {
    it('should simulate textChanged', async () => {
      const pwInput = testingLib.getByTestId('currentPwTextInput');
      const verifyBtn = testingLib.getByTestId('checkCurrentPwBtn');
      fireEvent.changeText(pwInput, 'left');

      fireEvent.press(verifyBtn);

      fireEvent.changeText(pwInput, 'right');
      fireEvent.press(verifyBtn);

      await wait(() => expect(testingLib.queryByTestId('newPwTextInput')).toBeTruthy());

      await testingLib.findByText(getString('CONFIRM'));

      const newPwInput = testingLib.getByTestId('newPwTextInput');
      const newPwInputCheck = testingLib.getByTestId('validationWordTextInput');
      fireEvent.changeText(newPwInput, 'test');
      fireEvent.changeText(newPwInputCheck, 'test1');
      fireEvent.press(verifyBtn);

      fireEvent.changeText(newPwInputCheck, 'test');
      fireEvent.press(verifyBtn);
    });
  });

  afterAll(() => {
    cleanup();
  });
});
