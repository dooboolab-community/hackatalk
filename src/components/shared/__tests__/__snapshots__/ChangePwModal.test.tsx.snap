// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[ChangePw] screen renders without crashing 1`] = `<View />`;

exports[`[ChangePw] screen renders without crashing 2`] = `
<View
  onLayout={[Function]}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
      "paddingBottom": 0,
      "paddingLeft": 0,
      "paddingRight": 0,
      "paddingTop": 20,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "height": 52,
            "justifyContent": "center",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "black",
              "fontSize": 20,
            },
          ]
        }
      >
        Change Password
      </Text>
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
        style={
          Array [
            Object {
              "position": "absolute",
              "right": 10,
            },
          ]
        }
        testID="closeBtn"
      >
        <View
          style={
            Object {
              "padding": 8,
            }
          }
        >
          <Text />
        </View>
      </TouchableOpacity>
    </View>
    <View
      style={
        Array [
          Object {
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "paddingBottom": 15,
            "paddingLeft": 15,
            "paddingRight": 15,
            "paddingTop": 15,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "flex-start",
              "alignSelf": "stretch",
              "flexDirection": "column",
              "justifyContent": "flex-start",
            },
          ]
        }
      >
        <Text
          focused={false}
          style={
            Array [
              Object {
                "color": "#a3a3a3",
                "fontSize": 12,
                "marginBottom": 8,
              },
            ]
          }
        >
          Enter Current Password
        </Text>
        <View
          isFocused={false}
          style={
            Array [
              Object {
                "alignItems": "center",
                "borderColor": "rgb(221,226,236)",
                "borderRadius": 3,
                "borderWidth": 1,
                "flexDirection": "row",
                "justifyContent": "center",
                "marginBottom": 8,
              },
            ]
          }
        >
          <TextInput
            allowFontScaling={true}
            autoCapitalize="none"
            autoCorrect={false}
            focused={false}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            placeholderTextColor="rgb(175,194,219)"
            rejectResponderTermination={true}
            secureTextEntry={true}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "alignSelf": "stretch",
                  "borderColor": "black",
                  "borderStyle": "solid",
                  "borderWidth": 0,
                  "color": "black",
                  "flexBasis": 0,
                  "flexDirection": "row",
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "fontSize": 16,
                  "justifyContent": "flex-start",
                  "paddingBottom": 22,
                  "paddingLeft": 20,
                  "paddingRight": 20,
                  "paddingTop": 22,
                },
              ]
            }
            testID="currentPwTextInput"
            textContentType="none"
            underlineColorAndroid="transparent"
          />
        </View>
      </View>
    </View>
    <TouchableOpacity
      activeOpacity={0.5}
      disabled={false}
      height={60}
      onPress={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "rgb(58,139,255)",
            "borderColor": "white",
            "borderRadius": 4,
            "borderWidth": 1,
            "height": 60,
            "justifyContent": "center",
            "width": "auto",
          },
        ]
      }
      testID="checkCurrentPwBtn"
      white={false}
      width="auto"
    >
      <Text
        disabled={false}
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 14,
              "fontWeight": "bold",
            },
          ]
        }
        white={false}
      >
        Next
      </Text>
    </TouchableOpacity>
  </View>
</View>
`;

exports[`[ChangePw] screen snapshot check if modal open 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
        "paddingBottom": 0,
        "paddingLeft": 0,
        "paddingRight": 0,
        "paddingTop": 20,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "height": 52,
              "justifyContent": "center",
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "black",
                "fontSize": 20,
              },
            ]
          }
        >
          Change Password
        </Text>
        <TouchableOpacity
          activeOpacity={0.2}
          style={
            Array [
              Object {
                "position": "absolute",
                "right": 10,
              },
            ]
          }
          testID="closeBtn"
        >
          <View
            style={
              Object {
                "padding": 8,
              }
            }
          >
            <Text />
          </View>
        </TouchableOpacity>
      </View>
      <View
        style={
          Array [
            Object {
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "paddingBottom": 15,
              "paddingLeft": 15,
              "paddingRight": 15,
              "paddingTop": 15,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "flex-start",
                "alignSelf": "stretch",
                "flexDirection": "column",
                "justifyContent": "flex-start",
              },
            ]
          }
        >
          <Text
            focused={false}
            style={
              Array [
                Object {
                  "color": "#a3a3a3",
                  "fontSize": 12,
                  "marginBottom": 8,
                },
              ]
            }
          >
            Enter Current Password
          </Text>
          <View
            isFocused={false}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderColor": "rgb(221,226,236)",
                  "borderRadius": 3,
                  "borderWidth": 1,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "marginBottom": 8,
                },
              ]
            }
          >
            <TextInput
              allowFontScaling={true}
              autoCapitalize="none"
              autoCorrect={false}
              focused={false}
              placeholderTextColor="rgb(175,194,219)"
              rejectResponderTermination={true}
              secureTextEntry={true}
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "alignSelf": "stretch",
                    "borderColor": "black",
                    "borderStyle": "solid",
                    "borderWidth": 0,
                    "color": "black",
                    "flexBasis": 0,
                    "flexDirection": "row",
                    "flexGrow": 1,
                    "flexShrink": 1,
                    "fontSize": 16,
                    "justifyContent": "flex-start",
                    "paddingBottom": 22,
                    "paddingLeft": 20,
                    "paddingRight": 20,
                    "paddingTop": 22,
                  },
                ]
              }
              testID="currentPwTextInput"
              textContentType="none"
              underlineColorAndroid="transparent"
            />
          </View>
        </View>
      </View>
      <TouchableOpacity
        activeOpacity={0.5}
        disabled={false}
        height={60}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "rgb(58,139,255)",
              "borderColor": "white",
              "borderRadius": 4,
              "borderWidth": 1,
              "height": 60,
              "justifyContent": "center",
              "width": "auto",
            },
          ]
        }
        testID="checkCurrentPwBtn"
        white={false}
        width="auto"
      >
        <Text
          disabled={false}
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 14,
                "fontWeight": "bold",
              },
            ]
          }
          white={false}
        >
          Next
        </Text>
      </TouchableOpacity>
    </View>
  </View>
</View>
`;
